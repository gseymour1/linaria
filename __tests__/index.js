/* @flow */

const babel = require('@babel/core');
const dedent = require('dedent');

it('should extract CSS to a comment', async () => {
  const { code } = await babel.transformAsync(
    dedent`
    var background = 'blue';

    const container = styled('div')\`
      background-color: ${'${background}'};
      color: ${'${props => props.color}'};
      width: ${'${100 / 3}'};
      border: 1px solid red;
    \`;
    `,
    {
      plugins: [require.resolve('../src/index')],
      filename: '/app/index.js',
    }
  );

  expect(code).toMatchSnapshot();
});
