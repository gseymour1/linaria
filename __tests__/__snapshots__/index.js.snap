// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should extract CSS to a comment 1`] = `
"var background = 'blue';
const container = component('div', \\"container_1ugh8t9\\", {
  \\"0\\": background,
  \\"1\\": props => props.color,
  \\"2\\": 100 / 3
});

/*CSS OUTPUT START

.container_1ugh8t9{background-color:var(--c-0);color:var(--c-1);width:var(--c-2);border:1px solid red;}.container_1ugh8t9:hover{border-color:blue;}

CSS OUTPUT END*/"
`;
